test/integration/apiserver/apply/reset_fields_test.go

```diff
@@ -85,60 +92,106 @@ func TestApplyResetFields(t *testing.T) {
 				t.Fatal(err)
 			}
 			t.Run(mapping.Resource.String(), func(t *testing.T) {
-				status, ok := statusData[mapping.Resource]
-				if !ok {
-					status = statusDefault
+				// TODO: delete this after updating test data for all resources
+				if mapping.Resource.Resource != "services" {
+					t.Skip()
 				}
-
-				newResource, ok := createData[mapping.Resource]
-				if !ok {
-					t.Fatalf("no test data for %s.  Please add a test for your new type to etcd.GetEtcdStorageData().", mapping.Resource)
-				}
-
-				newObj := unstructured.Unstructured{}
-				if err := json.Unmarshal([]byte(newResource.Stub), &newObj.Object); err != nil {
-					t.Fatal(err)
-				}
-				if err := json.Unmarshal([]byte(status), &newObj.Object); err != nil {
-					t.Fatal(err)
-				}
-
 				namespace := resetFieldsNamespace
 				if mapping.Scope == meta.RESTScopeRoot {
 					namespace = ""
 				}
-				name := newObj.GetName()
-				rsc := dynamicClient.Resource(mapping.Resource).Namespace(namespace)
-				_, err := rsc.Create(context.TODO(), &newObj, metav1.CreateOptions{FieldManager: "create_test"})
+
+				// create first object
+				obj1 := createObject(t, mapping, statusDefault, statusData, createData1)
+				obj1YAML, err := yaml.Marshal(obj1.Object)
 				if err != nil {
 					t.Fatal(err)
 				}
+				name := obj1.GetName()
 
-				statusObj := unstructured.Unstructured{}
-				if err := json.Unmarshal([]byte(status), &statusObj.Object); err != nil {
-					t.Fatal(err)
+				// apply the spec of the first object
+				apply1, err := dynamicClient.
+					Resource(mapping.Resource).
+					Namespace(namespace).
+					Patch(context.TODO(), obj1.GetName(), types.ApplyPatchType, obj1YAML, metav1.PatchOptions{FieldManager: "fieldmanager1"}, "")
+				if err != nil {
+					t.Fatalf("Failed to apply obj1: %v", err)
 				}
-				statusObj.SetAPIVersion(mapping.GroupVersionKind.GroupVersion().String())
-				statusObj.SetKind(mapping.GroupVersionKind.Kind)
-				statusObj.SetName(name)
-				statusYAML, err := yaml.Marshal(statusObj.Object)
+				fmt.Printf("apply1 = %+v\n", apply1)
+
+				// create second object
+				obj2 := createObject(t, mapping, resetFieldsStatusDefault, resetFieldsStatusData, createData2)
+				obj2YAML, err := yaml.Marshal(obj2.Object)
 				if err != nil {
 					t.Fatal(err)
 				}
 
-				// an unforced apply should not conflict if the resetfields are set correctly
-				_, err = dynamicClient.
+				// apply the status of the second object
+				// this won't conflict if resetfields are set correctly
+				// and will conflict if they are not
+				apply2, err := dynamicClient.
 					Resource(mapping.Resource).
 					Namespace(namespace).
-					Patch(context.TODO(), name, types.ApplyPatchType, statusYAML, metav1.PatchOptions{FieldManager: "apply_status_test"}, "status")
+					Patch(context.TODO(), name, types.ApplyPatchType, obj2YAML, metav1.PatchOptions{FieldManager: "fieldmanager2"}, "status")
```

Comment:
PatchStatus?

Reply:
oh nvm, is that what the `"status"` at the end is for?

Reply:
I was unaware of this, because the previous test did not use PatchStatus. I'm trying to find the docs for this now, but do you know if PatchStatus works on unstructured types?

Reply:
yes you are correct, that is what "status" at the end is for. (I commented before seeing your second comment)

Reply:
I actually don't know for sure that PatchStatus is a thing 😅 , I just didn't see any reference to status at all at first...